### ISSUE TITLE
"Report SPAM" option takes long to reload

### ISSUE THREAD

**Issue by yogev1978** [ScDor/alspam#65]
"Report SPAM" option rescans my messages every few minutes, even though in version 0.026 it shouldn't do so. Rescan takes 28 seconds in my case, which is very annoying.
![Screenshot_20201011-003830_אל-ספאם.jpg](https://user-images.githubusercontent.com/10349199/95666099-4f969780-0b5f-11eb-9f0d-f79fd714ad27.jpg)

**Comment by ScDor** [comment]
funny, it seems to rather be the contact reading. looks like we need another DAO object

```10-12 17:55:42.933 10336 10377 D SmsReader: reading messages with date > 1602478156053 (-1 if NA)
10-12 17:55:42.934 10336 10377 D SmsReader: constructed query for SMS db, query = date > 1602478156053
10-12 17:55:42.934 10336 10377 D SmsReader: querying SMS from phone, query = date > 1602478156053
10-12 17:55:42.935 10336 10377 D SmsReader: set last read time to LastSmsReadDateKey
10-12 17:55:42.937 10336 10336 D ViewRootImpl@735577b[ManualReportActivity]: dispatchAttachedToWindow
10-12 17:55:42.962 10336 10336 V Surface : sf_framedrop debug : 0x4f4c, game : false, logging : 0
10-12 17:55:42.963 10336 10336 D ViewRootImpl@735577b[ManualReportActivity]: Relayout returned: old=[0,0][0,0] new=[0,0][1080,2220] result=0x7 surface={valid=true 535281561600} changed=true
10-12 17:55:42.972 10336 10374 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, egl_color_buffer_format *, EGLBoolean) returns 0x3000,  [1080x2220]-format:1
10-12 17:55:42.972 10336 10374 D OpenGLRenderer: eglCreateWindowSurface = 0x7cc582a430
10-12 17:55:42.991 10336 10341 I zygote64: Do full code cache collection, code=100KB, data=111KB
10-12 17:55:42.991 10336 10341 I zygote64: After code cache collection, code=99KB, data=85KB
10-12 17:55:43.003 10336 10336 D ViewRootImpl@735577b[ManualReportActivity]: MSG_RESIZED_REPORT: frame=Rect(0, 0 - 1080, 2220) ci=Rect(0, 72 - 0, 144) vi=Rect(0, 72 - 0, 144) or=1
10-12 17:55:43.003 10336 10336 D ViewRootImpl@735577b[ManualReportActivity]: MSG_WINDOW_FOCUS_CHANGED 1
10-12 17:55:43.006 10336 10336 V InputMethodManager: Starting input: tba=android.view.inputmethod.EditorInfo@d3cfadc nm : org.alspam.app ic=null
10-12 17:55:43.006 10336 10336 I InputMethodManager: startInputInner - mService.startInputOrWindowGainedFocus
10-12 17:55:43.013 10336 10336 W StaticLayout: maxLineHeight should not be -1.  maxLines:1 lineCount:1
10-12 17:55:43.066 10336 10374 D OpenGLRenderer: eglDestroySurface = 0x7cc58287c0
10-12 17:55:43.078 10336 10336 D ViewRootImpl@5439828[MainActivity]: Relayout returned: old=[0,0][1080,2220] new=[0,0][1080,2220] result=0x5 surface={valid=false 0} changed=true
10-12 17:55:43.085 10336 10377 D SmsReader: done querying, read 8 messages
10-12 17:55:43.086 10336 10377 D SmsReader: filtering out messages from contacts
10-12 17:55:43.087 10336 10377 D ContactGetter: reading contacts from phone and caching
10-12 17:55:43.133 10336 10377 D ContactGetter: querying phone contact db
10-12 17:55:51.796 10336 10341 I zygote64: Do partial code cache collection, code=125KB, data=84KB
10-12 17:55:51.797 10336 10341 I zygote64: After code cache collection, code=125KB, data=84KB
10-12 17:55:51.797 10336 10341 I zygote64: Increasing code cache capacity to 512KB
10-12 17:56:12.523 10336 10377 D ContactGetter: got 1427 contact numbers
10-12 17:56:12.541 10336 10377 D DenyListHandler: queried denyList from local db (size=19 keys)
10-12 17:56:12.547 10336 10377 D SmsReader: filtered 1 messages out, returning 7 messages
10-12 17:56:12.547 10336 10377 I localSmsDbHandler: ranking and parsing
10-12 17:56:12.585 10336 10377 D DictionaryHandler: queried dictionary from local db (size=16 keys)
10-12 17:56:12.586 10336 10377 I smsProcessor: got 7 smsData objects
10-12 17:56:12.773 10336 10377 I smsProcessor: done processing
10-12 17:56:12.773 10336 10377 I localSmsDbHandler: adding 7 sms to db, took 226
10-12 17:56:12.788 10336 10377 I localSmsDbHandler: done, now there are 2067 messages in local db
```

**Comment by ScDor** [comment]
solved in 45debecc8a074b1442067cffb71f258045798aca
